src/independentset/random_maximal_ind_set.jl

```diff
@@ -0,0 +1,100 @@
+"""
+    random_maximal_independent_set_it(g)
+
+### Implementation Notes
+Performs [Approximate Maximum Independent Set](https://en.wikipedia.org/wiki/Maximal_independent_set#Sequential_algorithm) once.
+Returns a vector of vertices representing the vertices in the independent set.
+### Performance
+O(|V|+|E|)
+### Approximation Factor
+max(degree(g))+1
+"""
+function random_maximal_independent_set_it(
+    g::AbstractGraph{T}
+    ) where T <: Integer 
+  
+    ind_set = Vector{T}()  
+    deleted = zeros(Bool, nv(g))
+    perm_v = Random.shuffle(collect(vertices(g)))
+
+    for v in perm_v
+        deleted[v] && continue
+        deleted[v] = true
+        push!(ind_set, v)
+        for u in neighbors(g, v)
+            deleted[u] = true
+        end
+    end
+
+    return ind_set
+end
+
+best_ind_set(is1::Vector{T}, is2::Vector{T}) where T <: Integer = size(is1)[1] > size(is2)[1] ? is1 : is2
```

Comment:
`length` - and isn't this basically the inverse of
```
best_dom_set(ds1::Vector{T}, ds2::Vector{T}) where T <: Integer = size(ds1)[1] < size(ds2)[1] ? ds1 : ds2
```
?

If so, rename the function to something like `smallest_set` and use it in both places. (The only place where it's not inverse is when the lengths are equal, but that's probably a workable edge case.)